packages/block-editor/src/components/block-navigation/index.js

```diff
@@ -35,10 +36,25 @@ function BlockNavigationList( {
 				const blockType = getBlockType( block.name );
 				const isSelected = block.clientId === selectedBlockClientId;
 
+				/* HTML equivalent code
+				 * document.getElementsByClassName("editor-block-navigation__container")[0].closest(".components-popover__content").scrollTop
+				 *		= document.getElementsByClassName("components-button is-selected")[0].offsetTop
+				 */
+
+				const blockEl = useRef( null );
+
+				// similar to componentDidMount and componentDidUpdate
+				useEffect( () => {
+					if ( isSelected ) {
+						blockEl.current.closest( '.components-popover__content' ).scrollTop = block.offsetTop;
+					}
+				} );
```

Comment:
@nickylimjj This is pretty old now, but maybe it could still be useful?

You are using `isSelected` inside `useEffect`. The problem may actually be that you should be adding the var as a dependency. This way the code only executes when the var changes.

```suggestion
				}, [ isSelected ] );
```

I believe there is an ESLint rule that catches this now.